src/transaction-runner.coffee

```diff
@@ -111,8 +110,8 @@ class TransactionRunner
 
   # The 'data' argument can be 'transactions' array or 'transaction' object
   runHooksForData: (hooks, data, legacy = false, callback) ->
-    if hooks? and Array.isArray hooks
-      logger.debug 'Running hooks...'
+    if hooks?.length
```

Comment:
This is unrelated change as well. When there are no hooks, this method should skip directly to callback, not to feed `async.timesSeries()` with zero... I'm also guilty here for not sending this separately.